controllers/storagecluster/provider_server.go

```diff
@@ -189,6 +214,35 @@ func (o *ocsProviderServer) ensureDeploymentReplica(actual, desired *appsv1.Depl
 	return nil
 }
 
+// getWorkerNodesInternalIPAddresses return slice of Internal IPAddress of worker nodes
+func (r *StorageClusterReconciler) getWorkerNodesInternalIPAddresses() ([]string, error) {
+
+	nodes := &corev1.NodeList{}
+
+	err := r.Client.List(context.TODO(), nodes)
+	if err != nil {
+		r.Log.Error(err, "Failed to list nodes")
+		return nil, err
+	}
+
+	nodeAddresses := []string{}
+
+	for _, node := range nodes.Items {
```

Comment:
I do not think NodeList.items contain node references, as such this for operation is creating deep copies of the nodes' CR structures. These structures are large and might affect memory and garbage collection. I will suggest to do the following instead:
```suggestion
	for i, _ := range nodes.Items {
		node := &range nodes.Items[i]
	
	
```